DailySelfie/app/src/main/java/com/teamagam/dailyselfie/LoadThumbnailTask.java

```diff
@@ -0,0 +1,63 @@
+package com.teamagam.dailyselfie;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.BitmapFactory.Options;
+import android.os.AsyncTask;
+import android.view.View;
+import android.widget.ImageView;
+
+import java.lang.ref.WeakReference;
+
+class LoadThumbnailTask extends LoadPictureTask {
+    private final int mImageSize;
+
+    LoadThumbnailTask(ImageView imageView, int imageSize) {
+        // Use a WeakReference to ensure the ImageView can be garbage collected
+        super(imageView);
+        mImageSize = imageSize;
+    }
+
+    @Override
+    protected Bitmap doInBackground(String... params) {
+        String path = params[0];
+        return decodeSampledBitmapFromFile(path, mImageSize, mImageSize);
+    }
+
+    private static Bitmap decodeSampledBitmapFromFile(String path, int requiredWidth, int requiredHeight) {
+
+        // First decode with inJustDecodeBounds=true to check dimensions
```

Comment:
This method is a somewhat long.
Each comment here can be replaced with extracted the following code block into a new method with appropriate name.